src/pybind/mgr/dashboard/controllers/iscsi.py

```diff
@@ -938,3 +954,23 @@ def validate_rest_api(gateways):
                                          '{}'.format(gateway),
                                      code='ceph_iscsi_rest_api_not_available_for_gateway',
                                      component='iscsi')
+
+
+def validate_auth(auth):
+    username_regex = re.compile(r'^[\w\.:@_-]{8,64}$')
+    password_regex = re.compile(r'^[\w@\-_\/]{12,16}$')
+    result = True
+
+    if auth['user'] or auth['password']:
+        result = result and bool(username_regex.match(auth['user']))
+        result = result and bool(password_regex.match(auth['password']))
+
+    if auth['mutual_user'] or auth['mutual_password']:
+        result = result and bool(username_regex.match(auth['mutual_user']))
+        result = result and bool(password_regex.match(auth['mutual_password']))
+        result = result and auth['user'] and auth['password']
+
+    if not result:
```

Comment:
Is it possible to authenticate without `user` and `password`? (`result = true` in line 962)

Reply:
This is not authentication, just validation of the users and passwords.
Since its not required that you define a user and pass, it should return true if none is sent.